src/spark/utils.py

```diff
@@ -113,6 +114,14 @@ def get_spark_session(
 
     spark_conf = _get_base_spark_conf(app_name, executor_cores, yarn)
     sc = {}
+    if os.environ.get('USE_KUBE_SPAWNER') == 'true':
```

Comment:
Hmm, your codebase so up to you, but I would make this an argument rather than pulling from env vars inside a fn

Reply:
I need to make `get_spark_session` backwards compatible. Users are currently calling `spark = get_spark_session()`- this needs to be working after updates without changes from user's end.